index.ts

```diff
@@ -1,49 +1,56 @@
-require("dotenv").config();
-const { Client, Intents } = require("discord.js");
-
-const config = { phrases: {} };
-config.phrases.intro = require("./assets/phrases/intro.json");
-config.phrases.welcoming = require("./assets/phrases/welcoming.json");
-
+import { GuildMember, Message, Client, Intents, TextChannel } from "discord.js";
+import "dotenv/config.js";
 const { DISCORD_TOKEN } = process.env;
 
-const client = new Client({
-  intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MEMBERS, Intents.FLAGS.GUILD_MESSAGES],
-});
+interface IPhrases {
+  intro: Array<string>;
+  welcoming: Array<string>;
+}
 
 // #geral
-const DESTINATION_CHANNEL_ID = "855861944930402344";
+const DESTINATION_CHANNEL_ID = "1019589159020675135";
 // #comandos-testes
 // const DESTINATION_CHANNEL_ID = '807190194268012554';
+const config = { phrases: {} as IPhrases };
+config.phrases.intro = require("./assets/phrases/intro.json");
+config.phrases.welcoming = require("./assets/phrases/welcoming.json");
 
+const client = new Client({
+  intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MEMBERS, Intents.FLAGS.GUILD_MESSAGES],
+});
 client.once("ready", async () => {
   console.log("Ready!");
 });
-
 client.login(DISCORD_TOKEN);
 
-const getRandomStringFromCollection = (collection) => collection[Math.floor(Math.random() * collection.length)].trim();
+const getRandomStringFromCollection = (collection: Array<string>) =>
+  collection[Math.floor(Math.random() * collection.length)].trim();
 
-const replacePlaceholders = (phrases, memberID) => {
+const replacePlaceholders = (phrases: string, memberID: string) => {
   const memberIdTag = `<@${memberID}>`;
   return phrases.replace("{MEMBER_ID}", memberIdTag);
 };
 
-const welcome = async (memberID) => {
+const welcome = (memberID: string) => {
   const introPhrase = getRandomStringFromCollection(config.phrases.intro).trim();
   const welcomingPhrase = getRandomStringFromCollection(config.phrases.welcoming).trim();
   const finalPhrase = replacePlaceholders(`${introPhrase}${welcomingPhrase}`, memberID);
   console.log(`[NEW JOIN] ${finalPhrase}`);
   return finalPhrase;
 };
 
-client.on("guildMemberAdd", async (member) => {
-  console.log("Member joined the server!");
+client.on("guildMemberAdd", async (member: GuildMember) => {
   const message = await welcome(member.id);
-  member.guild.channels.cache.get(DESTINATION_CHANNEL_ID).send(message);
+  const textChannelData = member.guild.channels.cache.get(DESTINATION_CHANNEL_ID);
+
+  console.log("Member joined the server!");
+  if (!textChannelData)
+    return console.log("There was an error finding the text channel data, cannot send welcome message.");
+
+  (textChannelData as TextChannel).send(message);
```

Comment:
Não testei, mas não poderíamos ter o textChannelData logo como TextChannel (44) ao invés de lhe termos que fazer o cast mais tarde (50)?

Reply:
Pelo que eu vi, não. Porque `textChannelData` pode ser undefined, por isso a verificação. E depois o typesript fica chateado porque não consegue converter de `any | undefined` para `TextChannel`

Reply:
@gustavos00 podes fazer `const textChannelData : TextChannel = member.guild.channels.cache.get(DESTINATION_CHANNEL_ID);` se não me engano

Reply:
> @gustavos00 podes fazer `const textChannelData : TextChannel = member.guild.channels.cache.get(DESTINATION_CHANNEL_ID);` se não me engano

![image](https://user-images.githubusercontent.com/73289907/190372687-86647209-040e-4a96-b7af-84dbb304d7a0.png)
O problema é que `textChannelData` pode ser undefined.

Reply:
Um if resolve isso

Reply:
> Um if resolve isso

Sim, e isso já está feito. Eu tenho é de declarar a variável, ela vem como `GuildBasedChannel | undefined` como tipos, faço a condição e depois altero o tipo dela. Não vejo como o fazer sem ser por esta ordem.

Reply:
Porque não tens ambas as possibilidades de tipos definidos desde início?

```suggestion
  const textChannelData: TextChannel | undefined = member.guild.channels.cache.get(DESTINATION_CHANNEL_ID);
```

Reply:
> Porque não tens ambas as possibilidades de tipos definidos desde início?



Porque o resultado desse `member.guild.channels.cache.get(DESTINATION_CHANNEL_ID)` é do tipo `GuildBasedChannel | undefined` e o TS "não" aceita dizeres que ele é do tipo `TextChannel`.